neard/src/main.rs

```diff
@@ -122,12 +124,40 @@ fn main() {
             let account_id = args.value_of("account-id");
             let test_seed = args.value_of("test-seed");
             let genesis = args.value_of("genesis");
+            let download = args.is_present("download-genesis");
+            let download_url = args.value_of("download-genesis-url");
             let num_shards = args
                 .value_of("num-shards")
                 .map(|s| s.parse().expect("Number of shards must be a number"))
                 .unwrap_or(1);
             let fast = args.is_present("fast");
-            init_configs(home_dir, chain_id, account_id, test_seed, num_shards, fast, genesis);
+
+            if args.is_present("download-genesis") || args.is_present("download-genesis-url") {
+                if let Some(path) = genesis {
+                    if Path::new(path).exists() {
```

Comment:
This check is unnecessarily complicated. We should just check that when `args.is_present("download-genesis") || args.is_present("download-genesis-url")` is true, `genesis` must be `None` and vice versa.

Reply:
So, we still need to pass genesis as a path to where to download thr genesis.json file. So, not sure what to do here, I can download it into /tmp but thats not great either or directly to ~/.near folder?

Reply:
I see. It should be downloaded to `home` folder (which is an argument that defaults to `~/.near`).